packages/shared/src/open-feature.ts

```diff
@@ -122,26 +122,26 @@ export abstract class OpenFeatureCommonAPI<P extends CommonProvider = CommonProv
       return this;
     }
 
-    // get the named emitter, or if this is the default provider, get all event emitters not associated with a provider
-    const emitters = clientName ? [this.getAndCacheEventEmitterForClient(clientName)] : this.getUnboundEmitters();
-
+    const emitters = this.getAssociatedEventEmitters(clientName);
+    
     if (typeof provider.initialize === 'function') {
       provider
         .initialize?.(this._context)
         ?.then(() => {
-          emitters.forEach((emitter) => {
+          // fetch the most recent event emitters, some may have been added during init
+          this.getAssociatedEventEmitters(clientName).forEach((emitter) => {
```

Comment:
This was the bug. We we're getting the event emitters before this block. Sometimes, after a provider had been set, but BEFORE the `initialize` was complete, a handler would be added. We wouldn't have it in the list to run (the `emitters` variable was stale).

The fix is to get the most up-to-date list of handlers in the `then`.

Reply:
Ah good point!